lib/Runtime/Types/ES5ArrayTypeHandler.cpp

```diff
@@ -271,25 +279,34 @@ namespace Js
             doneConversion = true;
 #endif
         }
+#ifdef _MSC_VER
         __finally
         {
             if (AbnormalTermination())
```

Comment:
@Yongqu can you review this part of the change please? Can we convert this to using the TryFinally primitive that @jianchun introduced? (That is, do we really need SEH here or can we use just regular C++ exception handling mechanisms)